manifests/plugin/virt.pp

```diff
@@ -8,7 +8,7 @@
   Optional[String] $block_device                                             = undef,
   Optional[String] $interface_device                                         = undef,
   Optional[Boolean] $ignore_selected                                         = undef,
-  Optional[Enum['none', 'name', 'uuid', 'metadata']] $plugin_instance_format = undef,
+  Optional[String[1]] $plugin_instance_format                                   = undef,
```

Comment:
I was just following the other plugins pretty much. It does take a specific set of strings (one or many of), but the other templates don't seem to cover that, so I wasn't looking to make it more complicated than the other templates that already exist :)

It can take one of, or multiple of, those values. For example see https://github.com/voxpupuli/puppet-collectd/blob/master/manifests/plugin/virt.pp#L12 which is the same set up as this where it can take one of, or multiple of, a specific list of parameters.

I don't know if the list changes between various collectd versions either, so that would also need to be accounted for.

Reply:
fair point, it might also change between versions and that's quite ugly to implement.